packages/adblocker/src/lists.ts

```diff
@@ -136,13 +138,24 @@ export function f(strings: TemplateStringsArray): NetworkFilter | CosmeticFilter
 export function parseFilters(
   list: string,
   config: Partial<Config> = new Config(),
-): { networkFilters: NetworkFilter[]; cosmeticFilters: CosmeticFilter[] } {
+): {
+  networkFilters: NetworkFilter[];
+  cosmeticFilters: CosmeticFilter[];
+  preprocessors: Preprocessor[];
+} {
```

Comment:
@remusao forgot what we've decided - in case engine does not `loadPreprocessors` should the affected filters be kept or dropped? I think I'm leaning to the latter. 

@seia-soto this decision will likely affect `getFilters` which may have to be responsible to dropping affected filters. 

Wonder what to do with filters in `else` blocks? without understanding the reason behind the preprocessor condition it is impossible to make a decision, so we should consider potential breakage to the only factor. In such case, it's probably better to drop.

Reply:
Dropping affected filters is the current behavior. There's completely no difference compared to before preprocessor addition. At the same time, this can help you understand why I added "preservedFilters".

Reply:
`else` blocks can be treated same as normal if definitions.

Reply:
Not sure if it's better but I like the idea of having `loadPreprocessors === false` mean that we don't do any parsing/interpretation of these special directives at all (that's the current behavior before this PR); this would mean that filters from the blocks are interpreted as normal filters. If we don't do that, I do not really see the point of having a flag to disable it, since even if it's disabled we still interpret the preprocessor blocks but ignore their content. When would we want that?

Reply:
I also don't see any use for now. Even if we want to ignore all contents in the preprocessor blocks, we can just pass an empty env map.